plugins/modules/data_inputs_networks.py

```diff
@@ -0,0 +1,251 @@
+          - >
+            Sets the source key initial value. The key is used during parsing/indexing, in particular to set
+            the source field during indexing. It is also the source field used at search time. As a convenience,
+            the chosen string is prepended with 'source::'.
+          - >
+            Note: Overriding the source key is generally not recommended. Typically, the input layer provides a
+            more accurate string to aid in problem analysis and investigation, accurately recording the file from
+            which the data was retrieved. Consider use of source types, tagging, and search wildcards before
+            overriding this value.
+        type: str
+      sourcetype:
+        description:
+          - Set the source type for events from this input.
+          - '"sourcetype=" is automatically prepended to <string>.'
+          - Defaults to audittrail (if signedaudit=True) or fschange (if signedaudit=False).
+        type: str
+      token:
+        description:
+          - Token value to use for SplunkTcpToken. If unspecified, a token is generated automatically. 
+        type: str
+      password:
+        description:
+          - Server certificate password, if any.
+          - Only for TCP SSL configuration.
+        type: str
+      require_client_cert:
+        description:
+          - Determines whether a client must authenticate. 
+          - Only for TCP SSL configuration.
+        type: str
+      root_ca:
+        description:
+          - Certificate authority list (root file).
+          - Only for TCP SSL configuration.
+        type: str
+      server_cert:
+        description:
+          - Full path to the server certificate.
+          - Only for TCP SSL configuration. 
+        type: str
+      cipher_suite:
+        description:
+          - Specifies list of acceptable ciphers to use in ssl.
+          - Only obtained for TCP SSL configuration present on device.
+        type: str
+
+  running_config:
+    description:
+    - The module, by default, will connect to the remote device and retrieve the current
+      running-config to use as a base for comparing against the contents of source.
+      There are times when it is not desirable to have the task get the current running-config
+      for every task in a playbook.  The I(running_config) argument allows the implementer
+      to pass in the configuration to use as the base config for comparison. This
+      value of this option should be the output received from device by executing
+      command.
+    type: str
+  state:
+    description:
+    - The state the configuration should be left in
+    type: str
+    choices:
+    - merged
+    - replaced
+    - overridden
+    - deleted
+    - gathered
+    - rendered
+    - parsed
+    default: merged
+
+author: Pranav Bhatt (@pranav-bhatt)
+"""
+
+EXAMPLES = """
+- name: Example adding data input monitor with splunk.es.data_input_monitor
```

Comment:
@pranav-bhatt can u plz expand the example with expected output as well, like we do [here](https://github.com/ansible-collections/cisco.asa/blob/main/plugins/modules/asa_acls.py#L543) with both `before` and `after` if possible for all the supported states.